test/common/http/http1/conn_pool_test.cc

```diff
@@ -42,10 +42,15 @@ namespace Http1 {
 class ConnPoolImplForTest : public ConnPoolImpl {
 public:
   ConnPoolImplForTest(Event::MockDispatcher& dispatcher,
-                      Upstream::ClusterInfoConstSharedPtr cluster)
+                      Upstream::ClusterInfoConstSharedPtr cluster,
+                      NiceMock<Event::MockTimer>* upstream_ready_timer)
       : ConnPoolImpl(dispatcher, Upstream::makeTestHost(cluster, "tcp://127.0.0.1:9000"),
                      Upstream::ResourcePriority::Default, nullptr),
-        mock_dispatcher_(dispatcher) {}
+        mock_dispatcher_(dispatcher), mock_upstream_ready_timer_(upstream_ready_timer) {
+    ON_CALL(*mock_upstream_ready_timer_, enableTimer(std::chrono::milliseconds(0)))
+        .WillByDefault(Invoke(
+            [&](const std::chrono::milliseconds&) { mock_upstream_ready_timer_->callback_(); }));
```

Comment:
The fact that this gets invoked inline is not an accurate view of how the real code will work. I would not add this ON_CALL, and explicitly fire the ready even when needed outside of the immediate call stack. I really this is tedious, but given how important it is that this code is correct and how we would like it to track the real code as much as possible, I think it's worth it.

Reply:
Done.